### ISSUE TITLE
Empty home screen

### ISSUE THREAD

**Issue by diraneyya** [santimar/traefik-home#69]
Here is a screenshot of the Traefik dashboard showing the services:
<img width="1477" alt="image" src="https://github.com/user-attachments/assets/a2c296c3-15ef-4e9c-b2ad-109fef8722be">

And here is the parsed (i.e. `docker compose config`) docker compose output for one of the services:
```yaml
name: flatnotes
services:
  flatnotes:
    environment:
      FLATNOTES_AUTH_TYPE: password
      FLATNOTES_PASSWORD: 5cBKHhP1GWDv8i
      FLATNOTES_SECRET_KEY: zjUEHMbj53I8No9RX9u0yc0hiXPyeL
      FLATNOTES_USERNAME: diraneyya
    image: dullage/flatnotes:latest
    init: true
    labels:
      traefik-home-hide: "false"
      traefik-home.alias: flatnotes
      traefik-home.icon: https://raw.githubusercontent.com/dullage/flatnotes/develop/docs/logo.svg
      traefik.enable: "true"
      traefik.http.routers.flatnotes.entrypoints: https
      traefik.http.routers.flatnotes.rule: Host(`flatnotes.meffis.repair`)
      traefik.http.routers.flatnotes.tls.certresolver: letsencrypt
    networks:
      proxy: null
    restart: unless-stopped
    volumes:
      - type: volume
        source: data
        target: /app/data
        volume: {}
networks:
  proxy:
    name: proxy
    external: true
volumes:
  data:
    name: flatnotes_data
```

Logs show that the `traefik-home` application is able to see the container events:
```
traefik-home-1  | 2024/09/18 16:29:33 Received event stop for container c5601d7f5736
traefik-home-1  | 2024/09/18 16:29:33 Received event die for container c5601d7f5736
traefik-home-1  | 2024/09/18 16:30:59 Received event start for container 56e93dd59185
traefik-home-1  | 2024/09/18 16:30:59 Contents of /usr/share/nginx/html/index.html did not change. Skipping notification ''
```

And inspecting the running container shows the labels:
```
[
    {
        "Id": "56e93dd59185423469409471d8c4efb4faa24bade3c95659d67d92094c33a4f6",
        "Created": "2024-09-18T16:30:58.803987691Z",
        "Path": "/entrypoint.sh",
        "Args": [],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 207926,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2024-09-18T16:30:59.529277501Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2024-09-18T16:31:59.730189623Z",
                        "End": "2024-09-18T16:31:59.790116126Z",
                        "ExitCode": 0,
                        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\"OK\"\r100     4  100     4    0     0    666      0 --:--:-- --:--:-- --:--:--   666\n"
                    }
                ]
            }
        },
        "Image": "sha256:54b38ce6663e88c23ce29aec4f869c4761cd527c0f7faa8b4424cdbd7ed8f253",
        "ResolvConfPath": "/var/lib/docker/containers/56e93dd59185423469409471d8c4efb4faa24bade3c95659d67d92094c33a4f6/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/56e93dd59185423469409471d8c4efb4faa24bade3c95659d67d92094c33a4f6/hostname",
        "HostsPath": "/var/lib/docker/containers/56e93dd59185423469409471d8c4efb4faa24bade3c95659d67d92094c33a4f6/hosts",
        "LogPath": "/var/lib/docker/containers/56e93dd59185423469409471d8c4efb4faa24bade3c95659d67d92094c33a4f6/56e93dd59185423469409471d8c4efb4faa24bade3c95659d67d92094c33a4f6-json.log",
        "Name": "/flatnotes-flatnotes-1",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "36c11cac54e4c2c2ccbc124d338d209b7924c68a5543c754f2417eaa204f42f7",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "ConsoleSize": [
                0,
                0
            ],
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "private",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": [],
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "Mounts": [
                {
                    "Type": "volume",
                    "Source": "flatnotes_data",
                    "Target": "/app/data",
                    "VolumeOptions": {}
                }
            ],
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware",
                "/sys/devices/virtual/powercap"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ],
            "Init": true
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/adc36feaf77f456828bacb71405c506e7a34f22dc129210e054fe42352d60d2f-init/diff:/var/lib/docker/overlay2/1f22a64d6566d1b3286a64a4dd6ed67e6d423ade86749f9c1d2c75a3652f46c6/diff:/var/lib/docker/overlay2/0d289a9bca34af3f6dbe80330db2eb95ef07a49e2611ab81df11a6eba3ac542a/diff:/var/lib/docker/overlay2/f90c34c1f6b8aa9b90c2fe97daba38fd00b4808dc999e43d15a40d89986ca8ab/diff:/var/lib/docker/overlay2/fdaff620a18fe5cefd9f302bf3341e3637b081e6ab8ba4c579a6c15602b0ca27/diff:/var/lib/docker/overlay2/2480c05f5afc582fef2640f3297fce0d4ba995e1fe27509f2a6c3e5468046e92/diff:/var/lib/docker/overlay2/9af3b088cdc7359d7d0b786160ffffbe812f117e3af391a98bcc85ae14789cd0/diff:/var/lib/docker/overlay2/cbb5d37d14f5d74db7656bf4b3935ad04fdb3e3399d59bbfae56ab9198571991/diff:/var/lib/docker/overlay2/25b4a9b9fd41d499007f018043f57fed565f8dd5e15939f874e6f311db681b90/diff:/var/lib/docker/overlay2/118ed4c2b9d954f1242c4dc991e0b6357bb8182fa5c585d17ae01335522c1dde/diff:/var/lib/docker/overlay2/b09584308c23db5b7374caf68e90a62ff052ae1e5789fb8dc991260a5a787756/diff:/var/lib/docker/overlay2/882c2b297d0354496882cc2caf7e4c5068d44aca594de89da6540c799c6e0178/diff:/var/lib/docker/overlay2/87c45fb5edc94e20777eb5e2f363937102ef76b41a5b6e44f5a16685633f9cb6/diff:/var/lib/docker/overlay2/bbd7ed12bbd05135a7455392215d0b5dd919692b552cdd26fb4dac947664e69c/diff:/var/lib/docker/overlay2/56d8dc568327e95032fc07f43803c81e5bb023e1cfdd80a3146873f4d725640f/diff:/var/lib/docker/overlay2/03bf33cc9e9457675b7475ab94ea74613124db8638a6ca5dce0fdd076b38b8fe/diff:/var/lib/docker/overlay2/c606e2be3ef167b5c325998d9d7b93a7da49b877d49301a4a820939582eedb4e/diff",
                "MergedDir": "/var/lib/docker/overlay2/adc36feaf77f456828bacb71405c506e7a34f22dc129210e054fe42352d60d2f/merged",
                "UpperDir": "/var/lib/docker/overlay2/adc36feaf77f456828bacb71405c506e7a34f22dc129210e054fe42352d60d2f/diff",
                "WorkDir": "/var/lib/docker/overlay2/adc36feaf77f456828bacb71405c506e7a34f22dc129210e054fe42352d60d2f/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "volume",
                "Name": "27264b74c5d959b866e1939c81ea626a69431c8b544b15b013449dcaedde67af",
                "Source": "/var/lib/docker/volumes/27264b74c5d959b866e1939c81ea626a69431c8b544b15b013449dcaedde67af/_data",
                "Destination": "/data",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "volume",
                "Name": "flatnotes_data",
                "Source": "/var/lib/docker/volumes/flatnotes_data/_data",
                "Destination": "/app/data",
                "Driver": "local",
                "Mode": "z",
                "RW": true,
                "Propagation": ""
            }
        ],
        "Config": {
            "Hostname": "56e93dd59185",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": true,
            "AttachStderr": true,
            "ExposedPorts": {
                "8080/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "FLATNOTES_USERNAME=diraneyya",
                "FLATNOTES_PASSWORD=5cBKHhP1GWDv8i",
                "FLATNOTES_SECRET_KEY=zjUEHMbj53I8No9RX9u0yc0hiXPyeL",
                "FLATNOTES_AUTH_TYPE=password",
                "PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "LANG=C.UTF-8",
                "GPG_KEY=A035C8C19219BA821ECEA86B64E628F8D684696D",
                "PYTHON_VERSION=3.11.9",
                "PYTHON_PIP_VERSION=24.0",
                "PYTHON_SETUPTOOLS_VERSION=65.5.1",
                "PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/66d8a0f637083e2c3ddffc0cb1e65ce126afb856/public/get-pip.py",
                "PYTHON_GET_PIP_SHA256=6fb7b781206356f45ad79efbb19322caa6c2a5ad39092d0d44d0fec94117e118",
                "PUID=1000",
                "PGID=1000",
                "EXEC_TOOL=gosu",
                "FLATNOTES_PORT=8080",
                "APP_PATH=/app",
                "FLATNOTES_PATH=/data"
            ],
            "Cmd": null,
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "/healthcheck.sh"
                ],
                "Interval": 60000000000,
                "Timeout": 10000000000
            },
            "Image": "dullage/flatnotes:latest",
            "Volumes": {
                "/data": {}
            },
            "WorkingDir": "/app",
            "Entrypoint": [
                "/entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "com.docker.compose.config-hash": "8116d406d477656c431ae99eb31fc71eaf99542fc2cd940ed033079a100f7720",
                "com.docker.compose.container-number": "1",
                "com.docker.compose.depends_on": "",
                "com.docker.compose.image": "sha256:54b38ce6663e88c23ce29aec4f869c4761cd527c0f7faa8b4424cdbd7ed8f253",
                "com.docker.compose.oneoff": "False",
                "com.docker.compose.project": "flatnotes",
                "com.docker.compose.project.config_files": "/Users/orwa/repos/selfhosted/self-hosted/flatnotes/docker-compose.yml,/Users/orwa/repos/selfhosted/self-hosted/flatnotes/docker-compose.traefik.yml",
                "com.docker.compose.project.working_dir": "/Users/orwa/repos/selfhosted/self-hosted/flatnotes",
                "com.docker.compose.service": "flatnotes",
                "com.docker.compose.version": "2.29.2",
                "traefik-home-hide": "false",
                "traefik-home.alias": "flatnotes",
                "traefik-home.icon": "https://raw.githubusercontent.com/dullage/flatnotes/develop/docs/logo.svg",
                "traefik.enable": "true",
                "traefik.http.routers.flatnotes.entrypoints": "https",
                "traefik.http.routers.flatnotes.rule": "Host(`flatnotes.meffis.repair`)",
                "traefik.http.routers.flatnotes.tls.certresolver": "letsencrypt"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "fb72dd423bc2dd63e2d51639747380197f54c083e2d72820a31174ae160ec335",
            "SandboxKey": "/var/run/docker/netns/fb72dd423bc2",
            "Ports": {
                "8080/tcp": null
            },
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "proxy": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "flatnotes-flatnotes-1",
                        "flatnotes"
                    ],
                    "MacAddress": "02:42:ac:12:00:03",
                    "DriverOpts": null,
                    "NetworkID": "36c11cac54e4c2c2ccbc124d338d209b7924c68a5543c754f2417eaa204f42f7",
                    "EndpointID": "779bfcd5675d7ef740176b4780188be6ab44d924f508b257dc9fd2166c39abb9",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.3",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "DNSNames": [
                        "flatnotes-flatnotes-1",
                        "flatnotes",
                        "56e93dd59185"
                    ]
                }
            }
        }
    }
]
```

Is this project compatible with Traefik 3.1.2, which is the one I am using?

Is there something that I am missing?

**Comment by diraneyya** [comment]
The empty home screen:
<img width="1389" alt="image" src="https://github.com/user-attachments/assets/f5fd5b59-489a-4c7f-a38e-369bf12e098d">

**Comment by santimar** [comment]
Hi @diraneyya, this project was built for Traefik 2 and I didn't test it with the new 3.0 release.
It doesn't depend too much on Traefik to create the web view, but I guess that something in the syntax has changed and the app is not parsing correctly the configuration.

I will take a look to see what's changed and I'll report back.

**Comment by diraneyya** [comment]
> Hi @diraneyya, this project was built for Traefik 2 and I didn't test it with the new 3.0 release. It doesn't depend too much on Traefik to create the web view, but I guess that something in the syntax has changed and the app is not parsing the configuration correctly.
> 
> I will take a look to see what's changed and I'll report back.

Thank you! I would very greatly appreciate this.

**Comment by santimar** [comment]
Ok, I've made some tests with Traefik v3 and it works on my side.

After looking at your configuration I think I found the problem.
You are using `traefik.http.routers.flatnotes.entrypoints: https` but at the moment only the `web` and `websecure` endpoints are supported, as stated in readme file

You should edit your config to use the `websecure` endpoint or wait for #67 to be merged

**Comment by diraneyya** [comment]
Oh! I see!

So the endpoints defined in Traefik's static configuration file HAVE TO BE CALLED `web` and `websecure`, rather than anything else! I got it now!

It works and I am beyond grateful! Thanks!
<img width="1364" alt="image" src="https://github.com/user-attachments/assets/aa8707e4-e8b3-483d-830c-a2f5dd742c88">

**Comment by santimar** [comment]
Glad to hear that!
Closing

**Comment by carlosgfranco** [comment]
> Oh! I see!
> 
> So the endpoints defined in Traefik's static configuration file HAVE TO BE CALLED `web` and `websecure`, rather than anything else! I got it now!
> 
> It works and I am beyond grateful! Thanks! <img alt="image" width="1364" src="https://private-user-images.githubusercontent.com/107997002/369722016-aa8707e4-e8b3-483d-830c-a2f5dd742c88.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzU3NTE4NzYsIm5iZiI6MTc3NTc1MTU3NiwicGF0aCI6Ii8xMDc5OTcwMDIvMzY5NzIyMDE2LWFhODcwN2U0LWU4YjMtNDgzZC04MzBjLWEyZjVkZDc0MmM4OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQwOVQxNjE5MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZmE2ZjE4ZDNiMDc4NjA2MWJhM2IzZjA0YjVkZTgxZjVkZmNjNDI0MmQzODM1ZDc0NWY1MzBjZjllZmQ1YjE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.iUTkd6cLcmE3LDTAefHKnMSqFyRrZtmzTig8XTi6jt0">

Is this still a thing? I'm investigating why my flatnotes is not working behind traefik and I also happen to have renamed the entrypoints (web->http // websecure->https)...

**Comment by santimar** [comment]
@carlosgfranco  yes, this requirement is still needed as stated in the readme.

i'll send an update to allow customization soon

**Comment by santimar** [comment]
@carlosgfranco  Implemented in #114 and available in `v1.6.0`
